3 - FightClub/Code/Objects/Fighter.cs

```diff
@@ -0,0 +1,73 @@
+﻿using ObjectOrientedProblems.Code.Enums;
+using ObjectOrientedProblems.Code.Interfaces;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace FightClub.Code.Objects
+{
+    class Fighter : IFighter
+    {
+        public int Health => _health;
+
+        public int Damage => _damage;
+
+        public FighterState State => _curStatus;
+
+        const int _maxhealth=10;
+        private int _health;
+        private int _damage;
+        private FighterState _curStatus;
+
+        public void DrinkPotion(IPotion potion)
+        {
+            if (_curStatus == FighterState.Dead)
+            {
+                return;
+            }
+            _health += potion.Healing;
+            _health = Math.Min(_health, _maxhealth);
```

Comment:
Clever way to do this! Bonus points. I would have just accepted an If statement!